src/mechanics/dofs/DofCalcContainer.h

```diff
@@ -0,0 +1,115 @@
+#pragma once
+
+#include <vector>
+#include <eigen3/Eigen/Core>
+#include "mechanics/dofs/DofType.h"
+
+namespace NuTo
+{
+
+//! @brief dof container that is also capable of performing calculations.
+//! @tparam T type of the data that supports all the arithmetic operations
+template <typename T>
+class DofCalcContainer
```

Comment:
If I understand this correctly, we now have `DofContainer` for things like nodes and elements, and `DofCalcContainer` for vectors and matrices, right? I find the names `DofMatrix` and `DofVector` are fine, but the base class…
> dof container that is also capable of performing calculations

sounds like it stores `DofType`s, and the name does too. I can't come up with a better name at the moment (`DofLinAlgBase`^^); I'll get back to you when I do.

Reply:
I have no idea either and thanks for pointing out. Since this is not the only case of unsuitable names, I created #124.